ci/ci

```diff
@@ -114,64 +199,152 @@ def read_from_file(file_path):
+            os._exit(1)
+        status = read_from_file(os.path.join(
+            ci_dirname, MACROS['CONFIG_DIR'], MACROS['STATUS_FILE']))
+        for service in services:
+            if str(status).find(service) != -1:
+                services.remove(service)
+                service_count -= 1
+                image = (service.split('-', 1)[1][:-2]).replace('-', ':')
+                image = image.replace("_", ".")
+                images_to_commit[image] = service
+    for key in images_to_commit:
         try:
-            build_id = 'docker-compose ps -q pbs-build'
+            build_id = 'docker-compose -f docker-compose.json ps -q ' + \
+                images_to_commit[key]
             build_id = run_cmd(build_id, True)
             build_id = build_id.split("'")[1]
             build_id = build_id[:12]
-            # shortening the build id to 12 characters as is displayed by 'docker ps'
-            # unlike 'docker-compose ps'  which shows full id
+            image_name = (str(key).replace(':', '-')
+                          ).replace('.', '_') + '-ci-pbs'
+            # shortening the build id to 12 characters as is displayed by
+            # 'docker ps' unlike 'docker-compose ps'  which shows full id
             cmd = 'docker commit '+build_id+' '+image_name+':latest'
             log_info(cmd)
             run_cmd(cmd)
+        except Exception as e:
+            log_error(e)
+        try:
             bad_images = "docker images -qa -f'dangling=true'"
             bad_images = run_cmd(bad_images, True)
             if bad_images != "b''":
                 bad_images = (bad_images.split("'")[1]).replace("\\n", " ")
-                print("The following untagged images will be removed -> " + bad_images)
+                print("The following untagged images will be removed -> " +
+                      bad_images)
                 cmd = 'docker rmi ' + bad_images
                 run_cmd(cmd)
         except Exception as e:
-            log_error(e)
+            log_warning(
+                "could not remove bad (dangling) images, \
+                please remove manually")
+            print(e)
+    return True
 
 
-def get_pkgs(platform):
-    build_cmd = '/src/build-pbs-packages.sh'
-    if platform != 'called':
-        try:
-            subprocess.run([pbs_dirname + '/ci', '--delete'])
-            subprocess.run([pbs_dirname + '/ci', '--start=' + platform])
-        except:
-            log_error('Failed to restart ci.')
-    log_info('The package build logs can be found in logs/pkglogs')
-    run_docker_cmd(
-        'docker-compose exec pbs-build bash -c \' ' + build_cmd + ' | tee /logs/pkglogs \' ')
+def create_ts_tree_json():
+    benchpress_opt = os.path.join(
+        ci_dirname, MACROS['CONFIG_DIR'], MACROS['BENCHPRESS_OPT_FILE'])
+    benchpress_value = read_from_file(benchpress_opt)
+    try:
+        cmd = '/src/etc/gen_ptl_json.sh "' + benchpress_value + '"'
```

Comment:
there are some hard coded values, should they become part of MACROS?

Reply:
My opinion is that they are not used multiple times; and are unlikely to change in the future. It also provides for better readability of code. MACROS have all those strings which are likely to change or have multiple reference in the file.